localstack/services/cloudformation/cloudformation_api.py

```diff
@@ -936,6 +936,12 @@ def get_template(req_params):
     stack = find_stack(stack_name)
     if cs_name:
         stack = find_change_set(stack_name=stack_name, cs_name=cs_name)
+    else:
+        user_agent = request.headers.get("User-Agent")
+        if "aws-cdk/2" in user_agent:
+            change_set_id = stack.describe_details().get("ChangeSetId")
```

Comment:
Good catch, thanks for digging into this @gwright99 ! 👍 

I feel like we should apply the change more broadly here, as it is pointing at a more general issue. The problem seems to be that we're returning the original raw template of the stack, even if the stack had change sets applied to it (in which case we should be returning the latest template corresponding to the last change set.

We could include a new `latest_template_raw(..)` method in class `Stack`, something like (please note that this is untested!):
```
class Stack:
    ...
    def latest_template_raw(self):
        if self.change_sets:
            return self.change_sets[-1]._template_raw
        return self._template_raw
```

.. and then return `stack.latest_template_raw()` (instead of `stack._template_raw`) in this method.

Thoughts @dominikschubert ?

Reply:
@whummer  Good point, I agree. Especially given the recent additional CloudFormation cleanups, we should spend a bit time here to continue the plan we lightly sketched out previously for introducing clearer boundaries and flows in the deployer and api.

Reply:
It may be unrelated, but I encountered another problem re: the cloudformation logic here: https://github.com/localstack/localstack/issues/5401 related to the deployer and api flows. 

I don't think this problem is necessarily related to the code change I suggested above but call it out just in case because I haven't yet been able to really grok the recursive logic in the deployer.

Reply:
@gwright99 Thanks for reporting. I'll be looking into this :+1:

Reply:
Apologies for the delay on this one @gwright99 . Could you please try making the change with `latest_template_raw(..)` as outlined above, so we can remove the workaround based on the `User-Agent` value? Thanks!

Reply:
I'd be happy to try but won't be able to get to it until this evening.

Reply:
I can confirm that `latest_template_raw()` method works - I'm able generate proper diffs with `cdklocal diff` now.

![image](https://user-images.githubusercontent.com/1309341/152897095-feb5446b-7755-4837-9010-42392aff7edd.png)